From: Egbert Eich <eich@suse.com>
Date: Wed Mar 24 17:59:11 2021 +0100
Subject: rdma_find_network_type(): return MV2_NETWORK_CLASS_UNKNOWN when dev_list is freed
Patch-mainline: Not yet
Git-commit: 53c0c41fbd3728afb32cce7f23f596877e270911
References: 

This makes rdma_open_hca() bail early, so it doesn't reach the
code which tries to free dev_list and avoids a double free.

Signed-off-by: Egbert Eich <eich@suse.com>
---
 src/mpid/ch3/channels/mrail/src/gen2/rdma_iba_priv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
diff --git a/src/mpid/ch3/channels/mrail/src/gen2/rdma_iba_priv.c b/src/mpid/ch3/channels/mrail/src/gen2/rdma_iba_priv.c
index 9eb565e..4795035 100644
--- a/src/mpid/ch3/channels/mrail/src/gen2/rdma_iba_priv.c
+++ b/src/mpid/ch3/channels/mrail/src/gen2/rdma_iba_priv.c
@@ -507,7 +507,7 @@ int rdma_find_network_type(struct ibv_device **dev_list, int num_devices,
             if (dev_list[i]) {
                 ibv_ops.free_device_list(dev_list);
             }
-            return 0;
+            return network_type;
         }
         if (ERROR == rdma_find_active_port(nic_context[i], NULL, &hca_rate)) {
             /* No active port, skip HCA */
